File: Example/FrameWorkDemo/ZallData/ZAWebViewController.m
//
//  JSCallOCViewController.m
//  ZallDataSDK
//
//  Created by guo on 16/9/6.
//  Copyright © 2015-2020 Zall Data Co., Ltd. All rights reserved.
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
//

#import "ZAWebViewController.h"
#import <ZallDataSDK/ZallDataSDK.h>
#import <WebKit/WebKit.h>

@interface ZAWebViewController ()<WKNavigationDelegate, WKUIDelegate>
@property (nonatomic, strong) WKWebView *webView;
@end
@implementation ZAWebViewController
- (void)viewDidLoad
{
    [super viewDidLoad];
    _webView = [[WKWebView alloc] initWithFrame:self.view.bounds];
    self.title = @"WebView";

    NSString *path = [[[NSBundle mainBundle] bundlePath]  stringByAppendingPathComponent:@"test.html"];
    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL fileURLWithPath:path]];

    _webView.UIDelegate = self;
    _webView.navigationDelegate = self;
    [self.webView loadRequest:request];
 
    [self.view addSubview:_webView];
 
}

- (void)webView:(WKWebView *)webView runJavaScriptAlertPanelWithMessage:(NSString *)message initiatedByFrame:(WKFrameInfo *)frame completionHandler:(void (^)(void))completionHandler {
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"提示" message:message?:@"" preferredStyle:UIAlertControllerStyleAlert];
    [alertController addAction:([UIAlertAction actionWithTitle:@"确认" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        completionHandler();
    }])];
    [self presentViewController:alertController animated:YES completion:nil];
}

- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
    
//    if ([[ZallDataSDK sharedInstance] showUpWebView:webView WithRequest:navigationAction.request]) {
//            decisionHandler(WKNavigationActionPolicyCancel);
//        return;
//
//    }
    if ([navigationAction.request.URL.absoluteString hasPrefix:@"sensorsanalytics:"]) {
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"提示" message:navigationAction.request.URL.absoluteString?:@"" preferredStyle:UIAlertControllerStyleAlert];
        [alertController addAction:[UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleCancel handler:nil]];
        [self presentViewController:alertController animated:YES completion:nil];
        decisionHandler(WKNavigationActionPolicyCancel);
        return;
    }
    decisionHandler(WKNavigationActionPolicyAllow);
}

 

-(void)dealloc{
    
}

@end

File: Example/FrameWorkDemo/ZallDataSwift/ZAWebViewController.swift
//
//  JSCallOCViewController.swift
//  ZallDataSwift
//
//  Created by guo on 2021/12/10.
//  Copyright © 2021 Zall Data Co., Ltd. All rights reserved.
//

import UIKit
import WebKit
import ZallDataSDK
class ZAWebViewController: UIViewController, WKUIDelegate, WKNavigationDelegate {
    
    lazy var webView:WKWebView = {
        let webView = WKWebView(frame: view.bounds)
        webView.uiDelegate = self
        webView.navigationDelegate = self
        view.addSubview(webView)
       return webView
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = .white
//
//        let url = Bundle.main.resourceURL!.appendingPathComponent("test2.html");
//
//        let request = URLRequest(url: url)
//        webView.load(request)
        // Do any additional setup after loading the view.
        
        if let url = Bundle.main.url(forResource: "test", withExtension: "html") {
            let request = URLRequest(url: url)
            self.webView.load(request)
        }
        
        
    }
    
    
    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
        if (ZallDataSDK.sharedInstance()?.showUpWebView(webView, with: navigationAction.request))! {
            decisionHandler(WKNavigationActionPolicy.cancel)
            return;
        }
        decisionHandler(WKNavigationActionPolicy.allow)
    }
    
    
    
    
     

}
